### PR TITLE
Add support for LAZYDESIGNERS/THE60/rev2

### PR DESCRIPTION
**Pull request by jackytrabbit**
Add support for LAZYDESIGNERS/THE60/rev2

## Description

1. Add support for lazydesigners/the60/rev2
2. Put THE60 Round 1 in the60/rev1 folder
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by jackytrabbit** [comment]
@drashna I checked others' work and fixed the issue. Now it's using rev1/rev2 for different version. Thank you!

File: keyboards/lazydesigners/the60/rev1/readme.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+A 60% custom PCB and case designed and produced by [LazyDesigners](http://lazydesigners.cn).  
+The PCB allows one to test a 40% layout in a 60% case.
+
+Keyboard Maintainer: QMK Community  
+Hardware Supported: THE60 PCB  
+Hardware Availability: Check for GBs on [Geekhack](https://geekhack.org) and on [LazyDesigner's homepage.](http://lazydesigners.cn)  
+
+Make example for this keyboard (after setting up your build environment):
+
+    make lazydesigners/the60:default:dfu
```
Comment (Erovia) [inline keyboards/lazydesigners/the60/rev1/readme.md:14]:
```suggestion
    make lazydesigners/the60/rev1:default
```

**Review by Erovia — CHANGES_REQUESTED** [review]
This is currently breaking the user workflow as `lazydesigners/the60` will no longer be a valid build target.

You can work this around by having the following files:

`keyboards/lazydesigners/the60/rules.mk` file with the following content:
```
DEFAULT_FOLDER = lazydesigners/the60/rev1
```

`keyboards/lazydesigners/the60/the60.h`:
```
#pragma once

#include "quantum.h"

#ifdef KEYBOARD_lazydesigners_the60_rev1
    #include "rev1.h"
#elif KEYBOARD_lazydesigners_the60_rev2
    #include "rev2.h"
endif
```

`keyboards/lazydesigners/the60/the60.c`:
```
#include "the60.h"
```

**Comment by jackytrabbit** [comment]
@Erovia Got it! I have added the file to make 'make lazydesigners/the60:default' work. As rev2 will be the main version that THE60 will be made, I pointed the default folder to rev2.  Thanks for helping me with this!

File: keyboards/lazydesigners/the60/rev1/rules.mk [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = atmel-dfu
```
Comment (drashna) [inline keyboards/lazydesigners/the60/rev1/rules.mk:12]:
```suggestion
BOOTLOADER = atmel-dfu
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by jackytrabbit** [comment]
@Erovia Deleted rules.mk, the60.c and the60.h. Tested and everything is OK.

**Review by Erovia — APPROVED** [review]

**Pull request merged**